structuralcodes/sections/_generic.py

```diff
@@ -478,6 +581,233 @@ def calculate_moment_curvature(
 
         return res
 
+    def calculate_interaction_domain(
+        self, theta: float = 0, n_axial: int = 100
+    ) -> s_res.NmInteractionDomain:
+        """Calculate the NM interaction domain.
+
+        Arguments:
+        theta (float, default = 0): inclination of n.a. respect to y axis
+        n_axial (int, default = 100): number of discretization for axial load
+
+        Return:
+        (NmInteractionDomain)
+        """
+        # Prepare the results
+        res = s_res.NmInteractionDomain()
+        res.theta = theta
+        res.n_axial = n_axial
+        # Create an array of axial loads
+        res.n = np.linspace(self.n_min, self.n_max, n_axial)
+        # Initialize the result's arrays
+        res.m_y = np.zeros_like(res.n)
+        res.m_z = np.zeros_like(res.n)
+        # Compute strength for given angle of NA
+        for i, n_i in enumerate(res.n):
+            res_bend_strength = self.calculate_bending_strength(
+                theta=theta, n=n_i
+            )
+            res.m_y[i] = res_bend_strength.m_y
+            res.m_z[i] = res_bend_strength.m_z
+
+        return res
+
+    def calculate_interaction_domain_2(
+        self, theta: float = 0, n_axial: int = 100
+    ) -> s_res.NmInteractionDomain:
+        """Calculate the NM interaction domain (attempt to make it faster).
+
+        Arguments:
+        theta (float, default = 0): inclination of n.a. respect to y axis
+        n_axial (int, default = 100): number of discretization for axial load
+
+        Return:
+        (NmInteractionDomain)
+        """
+        # Prepare the results
+        res = s_res.NmInteractionDomain()
+        res.theta = theta
+        # with this version really n_axial is not used unless we perform
+        # some resampling after computation?
+        res.n_axial = n_axial
+
+        # Get ultimate strain profiles for theta angle
+        strains = self._compute_ultimate_strain_profiles(theta=theta)
+
+        # integrate all strain profiles
+        forces = np.zeros_like(strains)
+        for i in range(strains.shape[0]):
+            N, My, Mz, tri = (
+                self.integrator.integrate_strain_response_on_geometry(
+                    geo=self.section.geometry,
+                    strain=strains[i, :],
+                    tri=self.triangulated_data,
+                )
+            )
```

Comment:
Here, I guess we could do like?

```python
for strain in strains:
    N, My, Mz, tri = (
        self.integrator.integrate_strain_response_on_geometry(
            ...
            strain=strain,
            ...
        )
    )
```

Reply:
I wasn't sure how the for .. in .. worked with a 2D array. I'v learnt something new! thanks!

Reply:
Got it. You might do:

```python
for i, strain in enumerate(strains):
    N, My, Mz, tri = (
        self.integrator.integrate_strain_response_on_geometry(
            ...
            strain=strain,
            ...
        )
    )
```

Reply:
yes I know that, but it seemed not advantagueos..is it better for performance in you opinion?

Reply:
I don't think the performance will be significantly improved, but I find the code slightly more readable 👍